### DISCUSSION TITLE
Using Text to mask out and reveal the HTML below the canvas

### DISCUSSION THREAD

**Discussion by iDVB** [pixijs/pixijs#10365] [Q&A]
We have a working example of this without using PixiJS that converts the text to an SVG and then masks the 100% width and height white background with that. This creates a text "cut out" or "punch threw" to see the HTML below.

We would love to use PixiJS instead for many of the Text layout features it has. However, we're unable to get this working in PixiJS.

Closest we've come is [masking things out IN canvas](https://jsfiddle.net/Hackerham/s7hh0590/3/). 
But not the very important effect of punching through the text to see the below HTML.

Things we've tried:
- PixiJS v4 and v8
- 2hrs of convo and trial and error w ChatGPT v4
- 1hr of convo and trial and error w [GPT PixiJS Expert](https://chat.openai.com/g/g-FLexbpq62-pixijs-expert)
- v4 trying to [reverse above example](https://jsfiddle.net/e1c8dw3v/7/) and getting odd greyscale version vs inverted cut out.

Expected:
- canvas white rect filling the screen
- that white rect is cut out in the shape of the text
- canvas background is transparent which shows the image in the HTML behind the canvas...

we do not want that image, video or in our case the lottie animation that is behind the canvas to be processed IN the canvas. This must be outside and under the canvas as an HTML layer.

![image](https://github.com/pixijs/pixijs/assets/189506/544e232e-8c6b-4d26-99c1-838b61f3f102)

**Comment by RecursiveVoid** [comment]
So basically we go back to 2018;  https://github.com/pixijs/pixijs/issues/4710 

Create a BitmapText  => https://pixijs.download/v7.x/docs/PIXI.BitmapText.html 
Then mask it.

**Reply by bigtimebuddy** [reply]
For v8, this is answered in #10420
